src/server/TCPConnection.cpp

```diff
@@ -122,7 +122,12 @@ void TCPConnection::HandleRead(const boost::system::error_code &ec) {
     std::getline(is, data);
 
     const std::string const_data = std::string(data);
-    server_->Handle(const_data, shared_from_this());
+
+    unsigned int seed = 0;
+    bool drop_message = rand_r(&seed) % 1000 + 1 <= server_->GetDropRate();
+    if (!drop_message) {
```

Comment:
What is the plan with dropping messages on the TCP version of the server? I don't think we want to drop on TCP because TCP guarantees delivery for us.

Reply:
Dropping messages is required because it is how we will simulate killing servers.